File: interpreter/typescript/developer-expression.ts
import {IJobExpression} from "./job-expression.interface";
import {Context} from "./context";

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
export class DeveloperExpression implements IJobExpression {
  public interpret(context: Context): void {
    if(context.formula.includes('G')) {
      context.totalPoint += 4000;
    }
  }
}

File: interpreter/go/developer-expression.go
package main

import "strings"

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
type developerExpression struct{}

func NewDeveloperExpression() *developerExpression {
	return &developerExpression{}
}

func (e *developerExpression) Interpret(ctx *Context) {
	if strings.Contains(ctx.Formula, "G") {
		ctx.TotalPoint += 4000
	}
}
